pkg/storage/rdbms/rdbms.go

```diff
@@ -10,28 +11,31 @@ type ModuleStore struct {
 	connectionName string // settings name from database.yml
 }
 
-// NewRDBMSStorage  returns an unconnected RDBMS Module Storage
-// that satisfies the Storage interface. You must call
-// Connect() on the returned store before using it.
-// connectionName
-func NewRDBMSStorage(connectionName string) *ModuleStore {
-	return &ModuleStore{
+// NewRDBMSStorage  returns a connected RDBMS Module Storage
+// that satisfies the Storage interface.
+func NewRDBMSStorage(connectionName string) (*ModuleStore, error) {
+	const op errors.Op = "rdbms.NewRDBMSStorageWithConn"
+
+	ms := &ModuleStore{
 		connectionName: connectionName,
 	}
+	err := ms.connect()
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	return ms, nil
 }
 
 // NewRDBMSStorageWithConn  returns a connected RDBMS Module Storage
-// that satisfies the Storage interface. You must call
-// Connect() on the returned store before using it.
-// connectionName
+// that satisfies the Storage interface.
 func NewRDBMSStorageWithConn(connection *pop.Connection) *ModuleStore {
```

Comment:
Not sure if this method should really stay here, it's only used in a test (in test_suite.go) not it sure if it's really needed

Reply:
@kteb could the test use some other kind of module store? if so, it's up to you whether you'd like to remove it in this PR or wait for a follow-up. I usually do things like that in a follow up for what it's worth

Reply:
Follow-up: https://github.com/gomods/athens/issues/491